packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-button-strategy.ts

```diff
@@ -60,6 +71,43 @@ export default class PaypalCommerceButtonStrategy implements CheckoutButtonStrat
         return Promise.resolve();
     }
 
+    private async _handleOnCancel(_data: OnCancelData) {
+        const lineItems = this._getLineItems();
+        const existingConsignments = this._store.getState().consignments.getConsignmentsOrThrow();
+        const { email } = this._store.getState().billingAddress.getBillingAddressOrThrow();
+        const { firstName, lastName, address1 } = existingConsignments?.[0].shippingAddress || {};
+        if (this._currentShippingAddress) {
+            const shippingAddress = {
+                ...this._currentShippingAddress,
+                firstName: firstName !== 'Fake' ? firstName : '',
+                lastName: lastName !== 'Fake' ? lastName : '',
+                address1: address1 !== 'Fake street' ? address1 : '',
+                email: email !== 'fake@fake.fake' ? email : '',
+            };
+            const consignment = {
+                shippingAddress,
+                lineItems,
+            };
+            await this._store.dispatch(this._billingAddressActionCreator.updateAddress(shippingAddress));
+            if (existingConsignments[0]) {
+                await this._store.dispatch(this._consignmentActionCreator.updateAddress(shippingAddress));
+            } else {
+                await this._store.dispatch(this._consignmentActionCreator.createConsignments([consignment]));
+            }
```

Comment:
In this case you can remove if statement and use only `await this._store.dispatch(this._consignmentActionCreator.updateAddress(shippingAddress))`, because it creates new consignment under the hood if `existingConsignments` is not available